### PR TITLE
feat(CL): define swap fee on cl pool and create msg

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

This PR defined swap fee field on the CL pool struct and the CL `MsgCreatePool`.

Adds validation and tests for validate basic and pool constructor.

## Brief Changelog

- define swap fee on pool
- define swap fee on `MsgCreatePool`
- refactor dependencies and tests to set the new field
- refactor CL pool creation helpers to set swap fee
- add validations to pool constructor and test
- add validations to `MsgCreatePool`'s `ValidateBasic`
- review and update other tests
- run `go work sync`
- run `go mod tidy`

## Testing and Verifying

This change added tests

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/concentrated-liquidity/pool_test.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func (s *KeeperTestSuite) TestInitializePool() {
 
 	// Create a concentrated liquidity pool with invalid tick spacing
 	invalidTickSpacing := uint64(0)
-	invalidConcentratedPool, err := clmodel.NewConcentratedLiquidityPool(2, ETH, USDC, invalidTickSpacing, DefaultExponentAtPriceOne)
+	invalidConcentratedPool, err := clmodel.NewConcentratedLiquidityPool(2, ETH, USDC, invalidTickSpacing, DefaultExponentAtPriceOne, sdk.ZeroDec())
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/pool_test.go:37]: nit: Can we use DefaultZeroSwapFee

File: x/gamm/keeper/pool_service_test.go [hunk 2]
```diff
@@ -284,15 +283,8 @@ func (suite *KeeperTestSuite) TestInitializePool() {
...
-					defaultPoolId,
-					ETH,
-					USDC,
-					defaultTickSpacing,
-					DefaultExponentAtPriceOne,
-				)
-				require.NoError(suite.T(), err)
-				return &clPool
+				clPool := suite.PrepareConcentratedPool()
+				return clPool
```
Comment (czarcas7ic) [inline x/gamm/keeper/pool_service_test.go:287]: optional nit
```suggestion
				return suite.PrepareConcentratedPool()
```

**Review by czarcas7ic — APPROVED** [review]
LGTM!

File: x/concentrated-liquidity/genesis_test.go [hunk 3]
```diff
@@ -63,6 +63,7 @@ func TestInitGenesis(t *testing.T) {
 	require.Equal(t, testGenesis.Params.String(), clParamsAfterInitialization.String())
 	clPoolsAfterInitialization, err := app.ConcentratedLiquidityKeeper.GetAllPools(ctx)
 	require.NoError(t, err)
+	require.GreaterOrEqual(t, len(clPoolsAfterInitialization), 0)
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply x/concentrated-liquidity/genesis_test.go:66]: changed

File: x/gamm/keeper/pool_service_test.go (hunk 2 above)
Comment (p0mvn) [reply x/gamm/keeper/pool_service_test.go]: fixed locally

**Comment by p0mvn** [comment]
Thanks for the quick review @czarcas7ic . Going to merge with one approval since this change is low-risk and mostly just boilerpalte

**Pull request merged**